lib/CL/devices/level0/level0-driver.cc

```diff
@@ -0,0 +1,2239 @@
+                        Size, PatternSize);
+  ze_result_t Res = zeCommandListAppendMemoryFill(
+      CmdListH, DstPtr, Pattern, PatternSize, Size, nullptr, 0, nullptr);
+  LEVEL0_CHECK_ABORT(Res);
+}
+
+bool Level0Queue::setupKernelArgs(ze_module_handle_t ModuleH,
+                                  ze_kernel_handle_t KernelH, cl_device_id Dev,
+                                  unsigned DeviceI, _cl_command_run *RunCmd) {
+  cl_kernel Kernel = RunCmd->kernel;
+  struct pocl_argument *PoclArg = RunCmd->arguments;
+
+  /****************************************************************************************/
+
+  /* static locals are taken care of in ZE compiler */
+  assert(Kernel->meta->num_locals == 0);
+
+  cl_uint i;
+  ze_result_t Res;
+  for (i = 0; i < Kernel->meta->num_args; ++i) {
+    if (ARG_IS_LOCAL(Kernel->meta->arg_info[i])) {
+      assert(PoclArg[i].size > 0);
+      Res = zeKernelSetArgumentValue(KernelH, i, PoclArg[i].size, NULL);
+      LEVEL0_CHECK_ABORT(Res);
+
+    } else if (Kernel->meta->arg_info[i].type == POCL_ARG_TYPE_POINTER) {
+      assert(PoclArg[i].size == sizeof(void *));
+
+      if (PoclArg[i].value == NULL) {
+        Res = zeKernelSetArgumentValue(KernelH, i, sizeof(void *), nullptr);
+      } else if (PoclArg[i].is_svm) {
+        void *MemPtr = PoclArg[i].value;
+        Res = zeKernelSetArgumentValue(KernelH, i, sizeof(void *), &MemPtr);
+      } else {
+        cl_mem arg_buf = (*(cl_mem *)(PoclArg[i].value));
+        pocl_mem_identifier *memid = &arg_buf->device_ptrs[Dev->global_mem_id];
+        void *MemPtr = memid->mem_ptr;
+        Res = zeKernelSetArgumentValue(KernelH, i, sizeof(void *), &MemPtr);
+      }
+      LEVEL0_CHECK_ABORT(Res);
+      /*
+      ze_memory_advice_t Adv = (pa[i].is_readonly ?
+                                  ZE_MEMORY_ADVICE_SET_READ_MOSTLY :
+                                  ZE_MEMORY_ADVICE_CLEAR_READ_MOSTLY);
+      zeCommandListAppendMemAdvise(CmdList, DevHandle, MemPtr, memid->size,
+      Adv);
+      */
+    } else if (Kernel->meta->arg_info[i].type == POCL_ARG_TYPE_IMAGE) {
+      assert(PoclArg[i].value != NULL);
+      assert(PoclArg[i].size == sizeof(void *));
+
+      cl_mem arg_buf = (*(cl_mem *)(PoclArg[i].value));
+      pocl_mem_identifier *memid = &arg_buf->device_ptrs[Dev->global_mem_id];
+      void *hImage = memid->extra_ptr;
+      Res = zeKernelSetArgumentValue(KernelH, i, sizeof(void *), &hImage);
+      LEVEL0_CHECK_ABORT(Res);
+    } else if (Kernel->meta->arg_info[i].type == POCL_ARG_TYPE_SAMPLER) {
+      assert(PoclArg[i].value != NULL);
+      assert(PoclArg[i].size == sizeof(void *));
+
+      cl_sampler sam = (cl_sampler)(PoclArg[i].value);
+      ze_sampler_handle_t hSampler =
+          (ze_sampler_handle_t)sam->device_data[Dev->dev_id];
+
+      Res = zeKernelSetArgumentValue(KernelH, i, sizeof(void *), &hSampler);
+      LEVEL0_CHECK_ABORT(Res);
+    } else {
+      /* Normally plain-old-data arguments are passed into the kernel via a
```

Comment:
This file should follow the C+ guideliness, thus a C++ comment style //.

Reply:
fixed